README.md

```diff
@@ -40,99 +41,159 @@ This package also provides optimized functions to compute column-wise and pairwi
+For instance, in Julia v1.1 and above, the colwise behavior shown above can be
+equally well obtained by
+
+```julia
+r = colwise(dist, eachcol(X), eachcol(Y))
+```
+
+Equally simple, the analogous "rowwise" distance computation can be obtained via
+
+```julia
+r = colwise(dist, eachrow(X), eachrow(Y))
+```
 
-Note that either of `X` and `Y` can be just a single vector -- then the `colwise` function will compute the distance between this vector and each column of the other parameter.
+The iterator-based approach is in fact the future-proof way of computing distances
+between corresponding data points of iterable collections `X` and `Y`. Exceptions
+are given by distance types with `colwise` methods specialized for
+`AbstractVecOrMat` arguments, i.e., `CorrDist` and `[Sq]Mahalanobis`.
 
-#### Computing pairwise distances
+### Computing pairwise distances
 
-Let `X` and `Y` respectively have `m` and `n` columns. Then the `pairwise` function with the `dims=2` argument computes distances between each pair of columns in `X` and `Y`:
+Let `X` and `Y` have `m` and `n` columns, respectively, and the same number of rows.
+Then the `pairwise` function with the `dims=2` argument computes distances between
+each pair of columns in `X` and `Y`:
 
 ```julia
 R = pairwise(dist, X, Y, dims=2)
 ```
 
-In the output, `R` is a matrix of size `(m, n)`, such that `R[i,j]` is the distance between `X[:,i]` and `Y[:,j]`. Computing distances for all pairs using `pairwise` function is often remarkably faster than evaluting for each pair individually.
+In the output, `R` is a matrix of size `(m, n)`, such that `R[i,j]` is the
+distance between `X[:,i]` and `Y[:,j]`. Computing distances for all pairs using
+`pairwise` function is often remarkably faster than evaluting for each pair
+individually.
 
-If you just want to just compute distances between columns of a matrix `X`, you can write
+If you just want to just compute distances between all columns of a matrix `X`,
+you can write
 
 ```julia
 R = pairwise(dist, X, dims=2)
 ```
 
-This statement will result in an `m-by-m` matrix, where `R[i,j]` is the distance between `X[:,i]` and `X[:,j]`.
-`pairwise(dist, X)` is typically more efficient than `pairwise(dist, X, X)`, as the former will take advantage of the symmetry when `dist` is a semi-metric (including metric).
+This statement will result in an `m-by-m` matrix, where `R[i,j]` is the distance
+between `X[:,i]` and `X[:,j]`. `pairwise(dist, X)` is typically more efficient
+than `pairwise(dist, X, X)`, as the former will take advantage of the symmetry
+when `dist` is a semi-metric (including metric).
+
+For performance reasons, it is recommended to use matrices with observations in
+columns (as shown above). Indeed, the `Array` type in Julia is column-major,
+making it more efficient to access memory column by column. However, matrices
+with observations stored in rows are also supported via the argument `dims=1`.
```

Comment:
BTW, I've never been able to reproduce this alleged performance difference. Do you have any idea how we could reproduce it?

Reply:
No idea. We could leave out the recommendation part and simply mention the row case, without judgement.

Reply:
Yeah I'm not sure what's best. One day we should make more serious benchmarks to resolve this question. :-)